src/util/strencodings.cpp

```diff
@@ -107,18 +107,17 @@ std::vector<unsigned char> ParseHex(const std::string& str)
     return ParseHex(str.c_str());
 }
 
-void SplitHostPort(std::string in, uint16_t& portOut, std::string& hostOut)
+void SplitHostPort(std::string in, std::string& portOut, std::string& hostOut)
```

Comment:
I don't think we gain anything by having a string-only parser?

Doing this means `SplitHostPort` will only conditionally validate port range.